### ISSUE TITLE
Add flag to skip git init + git add

### ISSUE THREAD

**Issue by kallisti5** [haikuports/haikuporter#72]
```
13:22 <@PulkoMandy> we could probably run haikuporter without git on the slaves                                                                                                                             
13:22 <@PulkoMandy> (using plain extract + patch)                                                                                                                                                           
13:22 <@PulkoMandy> the git is only useful for --extract-patchset      
```

**Comment by kallisti5** [comment]
Some large (LARGE) projects, the git add takes a long, long time. (webkit 5+ hours).  haikuporter could use a flag to skip the git repo creation process.

**Comment by jessicah** [comment]
Looking at the pull request, this would only work if there are no patches to apply, which very much limits the utility of this option.

Shouldn't it instead apply patches with patch instead of git am?

**Comment by KapiX** [comment]
OK, `patch` does work with git patchsets. I thought it does not. @DarkmatterVale can you rework your solution?

**Comment by DarkmatterVale** [comment]
@KapiX Yes, I am trying to find a solution right now.

**Comment by KapiX** [comment]
Fixed by 7e5661a994db976ae897e2ef42b07382cb1699b2.
